Justin/handle user ticket

jtyj-github opened the pull request:
- Add api to store user ticket to prisma
- Add getUserTicket api

User Ticket is stored as such (id is unique identifier for each **ticket**, userId is unique identifier for each **user**):
![image](https://github.com/user-attachments/assets/94280651-a23e-4b97-b719-e5aa7a9fe852)

Commit:
🔧 chore: remove redundant field in totoTicket model

Commit:
🚀 perf: add type guard to notificationList

Commit:
⚡️ feat: add storeUserTicket api

Commit:
🔧 chore: edit totoTicket prisma model

Commit:
🧪 test: add code for testing storeUserTicket functionality

Commit:
🔧 chore: update totoTicket model to include uuid

Commit:
🔧 chore: included uuid

Commit:
⚡️ feat: add getUserId util

Commit:
⚡️ feat: add getUserTicket api

Commit:
🔧 chore: include uuid param to be stored in db

Commit:
🔧 chore: remove date param from totoTicket

Commit:
🔧 chore: next steps for getUserId

Commit:
🔧 chore: remove date from totoTicket creation

andrehadianto committed:
Merge remote-tracking branch 'origin/main' into justin/handle-user-ticket

andrehadianto committed:
🔧 chore: add package (uuid)

andrehadianto committed:
♻️ refactor: getUserId to become a hook used in client

andrehadianto committed:
♻️ refactor: moves to app/api page. convert to route

andrehadianto commented on src/app/api/tickets/route.ts:
```diff
@@ -0,0 +1,23 @@
+    try {
+        const userTickets = await prisma.totoTicket.findMany({
+            where: {
+                uuid: userId
```
there's red squiggly on this `uuid`. changing it to `id` seems to fix it. is there a reason for using `uuid`? or is there a way to make `uuid` the default?

jtyj-github replied:
There isnt a red squiggly on my side, but the uuid is the field defined in the prisma schema. This issue could be because the prisma schema was updated in the schema, but not in you local env. 
 
Maybe running 'npx prisma generate' on your end will rectify this issue

Commit:
♻️ refactor: remove additionalNum for submitted ticket

The user's ticket will not contain an additional number, this number is only known from the draw

andrehadianto committed:
♻️ refactor: use next14 new request type

Commit:
🧪 test: api/tickets/store api test code

Commit:
🔧 chore: Resolved merge conflicts

Commit:
🔧 chore: Resolved merged conflicts

Commit:
♻️ refactor: Use next14 new request type

Commit:
♻️ refactor: removed NextApiResponse import

Commit:
🐛 fix: wrong folderLocation being selected

Commit:
⚡️ feat: Add api route for retrieving toto Cards

Commit:
♻️ refactor: Change winningShares to winningPool for standardization

Commit:
🔧 chore: Simplified response to a single message

Commit:
♻️ refactor: follow same naming convention as scraped data

Commit:
♻️ refactor: obtain toto cards from api route

Commit:
♻️ refactor: moved useUser into page.tsx

andrehadianto committed:
🐛 fix: linting

andrehadianto committed:
⚡️ feat: add toto submission modal

andrehadianto committed:
♻️ refactor: integrate modal to homepage

andrehadianto committed:
♻️ refactor: reset value after submission
